Objective-C/LogSink/CBLConsoleLogSink.mm

```diff
@@ -0,0 +1,76 @@
+//
+//  CBLConsoleLogSink.m
+//  CouchbaseLite
+//
+//  Copyright (c) 2024 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "CBLConsoleLogSink.h"
+#import "CBLLogSinks+Internal.h"
+
+static NSArray* logLevelNames = @[@"Debug", @"Verbose", @"Info", @"WARNING", @"ERROR", @"none"];
+
+@implementation CBLConsoleLogSink
+
+@synthesize level=_level, domain=_domain;
+
+- (instancetype) initWithLevel: (CBLLogLevel)level {
+    return [self initWithLevel: level domain: kCBLLogDomainAll];
+}
+
+- (instancetype) initWithLevel: (CBLLogLevel)level domain: (CBLLogDomain)domain {
+    self = [super init];
+    if (self) {
+        _level = level;
+        _domain = domain;
+    }
+    return self;
+}
+
+- (void) writeLogWithLevel: (CBLLogLevel)level domain: (CBLLogDomain)domain message: (NSString*)message { 
+    if (level < self.level || (self.domain & domain) == 0) {
+        return;
+    }
+    
+    NSString* levelName = logLevelNames[level];
+    NSString* domainName = [self domainName: domain];
+    NSLog(@"CouchbaseLite %@ %@: %@", domainName, levelName, message);
+}
+
```

Comment:
Isn't this an attribute of the LiteCore interface?  Should it be somewhere generic for use by all of the loggers?

Reply:
Not sure what you mean here? NSLog is an Obj-C native method to write to console only.